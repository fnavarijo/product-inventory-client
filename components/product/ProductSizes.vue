<template>
  <div class="my-4 flex">
    <span class="mr-4">Tallas</span>
    <span v-for="(size, index) in sizes" :key="index" class="mx-1 font-bold">{{
      size
    }}</span>
  </div>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue';

export default Vue.extend({
  name: 'ProductSizes',
  props: {
    sizes: {
      type: Array as PropType<Array<string>>,
      default: () => [],
    },
  },
});
</script>
