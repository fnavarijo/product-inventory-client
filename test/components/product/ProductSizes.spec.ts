import { mount } from '@vue/test-utils';

import ProductSizes from '@/components/product/ProductSizes.vue';

describe('ProductSizes component', () => {
  it('should generate sizes component with correct style', () => {
    const wrapper = mount(ProductSizes, {
      propsData: { sizes: ['S', 'L', 'XL'] },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
