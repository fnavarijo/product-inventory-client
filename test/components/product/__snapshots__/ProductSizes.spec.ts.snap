// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSizes component should generate sizes component with correct style 1`] = `<div class="my-4 flex"><span class="mr-4">Tallas</span> <span class="mx-1 font-bold">S</span><span class="mx-1 font-bold">L</span><span class="mx-1 font-bold">XL</span></div>`;
